### PR TITLE
Add from_v1 classmethod

### PR DESCRIPTION
**Pull request by dchirst**
Class method to convert a TMS 1.0 spec to a v2 TMS

- added a classmethod to the TileMatrixSet class
- Added tests to make sure it works

### CODE REVIEW

**Comment by dchirst** [comment]
@vincentsarago Is it okay if you review it here? We don't want to merge it into devseed's repo atm so this is the best idea I can think of.

**Comment by vincentsarago** [comment]
Oh yeah you can't easily make a PR on top of another of your PR easily on fork 😓 

no problem to me 👍

File: morecantile/models.py [hunk 1]
```diff
@@ -233,21 +233,60 @@ def _invert_axis(self) -> bool:
         """Check if CRS has inverted AXIS (lat,lon) instead of (lon,lat)."""
         return crs_axis_inverted(self.crs)
 
+    @classmethod
+    def from_v1(cls, **kwargs) -> "TileMatrixSet":
```
Comment (vincentsarago) [inline morecantile/models.py]: what the advantage of using `**kwargs: Any` over `tms: Dict` ?

File: tests/test_models.py [hunk 2]
```diff
@@ -364,6 +366,26 @@ def test_mars_local_tms():
     assert round(center.y, 6) == 17
 
 
+@pytest.mark.parametrize(
+    "identifier, url, crs", [
+        ("UPSAntarcticWGS84Quad", "https://raw.githubusercontent.com/vincentsarago/TileMatrixSets/master/UPSAntarcticWGS84Quad.json", 5042),
+        ("CanadianNAD83_LCC", "https://raw.githubusercontent.com/vincentsarago/TileMatrixSets/master/CanadianNAD83_LCC.json", 3978),
+        ("WebMercatorQuad", "https://raw.githubusercontent.com/vincentsarago/TileMatrixSets/master/WebMercatorQuad.json", 3857)
+    ]
+)
```
Comment (vincentsarago) [inline tests/test_models.py:375]: better use local fixtures (to avoid fetching things from the internet)

File: morecantile/models.py (hunk 1 above)
Comment (dchirst) [reply morecantile/models.py]: None at all. Changed in current commit

File: tests/test_models.py (hunk 2 above)
Comment (dchirst) [reply tests/test_models.py:375]: Changed!

File: morecantile/models.py [hunk 3]
```diff
@@ -233,21 +233,63 @@ def _invert_axis(self) -> bool:
...
-        tile_height: int = 256,
-        matrix_scale: Optional[List] = None,
-        extent_crs: Optional[CRS] = None,
-        minzoom: int = 0,
-        maxzoom: int = 24,
-        title: str = "Custom TileMatrixSet",
-        id: str = "Custom",
-        ordered_axes: Optional[List[str]] = None,
-        geographic_crs: CRS = WGS84_CRS,
+            cls,
```
Comment (vincentsarago) [inline morecantile/models.py:280]: not sure why there are changes here but it should get fixed by pre-commit once we merge in the other PR

**Review by vincentsarago — APPROVED** [review]

**Pull request merged**